pxr/usd/usdLux/shapingAPI.h

```diff
@@ -249,6 +322,37 @@ class UsdLuxShapingAPI : public UsdAPISchemaBase
     // --------------------------------------------------------------------- //
     /// An IES (Illumination Engineering Society) light
     /// profile describing the angular distribution of light.
+    /// 
+    /// For full details on the .ies file format, see the full specification,
+    /// ANSI/IES LM-63-19:
+    /// 
+    /// https://store.ies.org/product/lm-63-19-approved-method-ies-standard-file-format-for-the-electronic-transfer-of-photometric-data-and-related-information/
+    /// 
+    /// The luminous intensity values in the ies profile are sampled using
+    /// the emission direction in the light's local space (after a possible
+    /// transformtion by a non-zero shaping:ies:angleScale, see below). The
+    /// sampled value is then potentially normalized by the overal power of the
+    /// profile if shaping:ies:normalize is enabled, and then used as a scaling
+    /// factor on the returned luminance:
+    /// 
```

Comment:
We should probably also mention that IES profile's photometry may or may not be in relative to a benchmark "bare bulb", and so we expect that IES profiles should be considered to require scaling of some sort in any case, whether or not we are also imposing normalization.

Reply:
We're actually going to need to revisit this. We're getting into the weeds with this internally here and the correct handling for converting from intensities in the IES to exitant luminance from an area source is more involved than what's being done in any renderer currently.

Reply:
When I wrote that, I was trying to correlate the IES specification to what we're saying here. I had been thinking of the intensity pattern as if it is a modulating the emittance from a small spherical radiator. Since the pattern is measured from a goniophotometer, the measurement itself pretends that the luminaire is a sphere; so basically a sphere large enough to have a surface area that can be measured, but small enough that it can treated like a point light in terms of how the light looks like from two different viewpoints, ie, not considering a difference in irradiance across the span of the luminaire... When you say "from an area source", that's very interesting. Whenever I look at Annex D, I always wonder if the height, width, and length they talk about are angular or linear. Since no units are measured, I've always assumed angular, and assumed that there's no way to guess how big a luminaire is from an IES file. I'm very curious to learn how you're thinking about this, and if there is a clear explanation in LM-63-19 and I just haven't spotted it.

Reply:
Yeah so basically units in the IES are candela by definition. To compute the correct exitant luminance one needs to divide that by area of the source along the exitant direction. For a sphere this is always $\pi r^2$ since the sphere projects to a disk along any direction. For a disk light this would be the area of the disk, multiplied by the cosine of the angle between the exitant direction and the disk surface normal. For rectangles and cylinders it gets more complicated. 

I’m currently pondering what the best approach is here. IES files are only really valid for point lights by their nature. For spheres and disks we can do something reasonable and simple as noted above. For other shapes do we try and be exact (cylinders seem like they’d be particularly hard/expensive to handle)? Or do we just make some simple approximation and note that the output is not expected to be correct for those shapes. The latter seems reasonable to me but haven’t thought it through fully yet.

Reply:
Would be keen to know your thoughts

Reply:
Ah, I see where you are going with that, I think. I wonder if it might be easiest to think about by taking a step beyond a sphere, by considering one face of a 10cm x 10cm x 10cm cube, with a point emitter at 0,0,0. A disc projected from the emitter to the face would have an area of 100 cm^2. We could then just treat the face as a disc light using `L = I / area`.  To finish that thought, we could treat the whole ideal cube luminaire as six of those.

That wouldn't be a bad approximation, I bet.

If we thought about an icosahedron, with 20 disc lights, the approximation is better.

Does this reduce to a differential form, that we could plug the small set of regular IES shapes into? I think there's sphere, ellipsoid, box, cylinder, and not much else?

For a cylinder it might be $dA = _frac{ dI } { r d\theta dz cos(\theta ') $ on the walls, and a plain old disk for the top and bottom.

Reply:
To be clear I'm talking about the shapes of the usdlux lights, so sphere, disk, rect, cylinder